salt/grains/extra.py

```diff
@@ -21,7 +22,14 @@ def shell():
     '''
     # Provides:
     #   shell
-    return {'shell': os.environ.get('SHELL', '/bin/sh')}
+    if salt.utils.platform.is_windows():
+        env_var = 'COMSPEC'
+        default = r'C:\Windows\system32\cmd.exe'
```

Comment:
Maybe something like this os.getenv('ComSpec', '{0}\\system32\\cmd.exe'.format(os.getenv('WINDIR')))

Reply:
I did consider it, but thought if COMSPEC isn't set (system would be very broken) there is no guarantee that WINDIR is set either. So this way at least the path is valid, i.e. /bin/sh might not exist either.